src/resolve.ts

```diff
@@ -0,0 +1,15 @@
+import * as path from "path";
+import * as ts from "typescript";
+import { normalizeSlashes } from "./utils";
+
+export function getProjectRootDir(program: ts.Program): string {
+    const options = program.getCompilerOptions();
+    const projectDir = options.configFilePath || program.getCommonSourceDirectory();
+
+    return normalizeSlashes(options.rootDir ? path.resolve(projectDir, options.rootDir) : projectDir);
```

Comment:
When `rootDir` is specified it always should be resolved from config file, common source directory should be used as a fallback only when it's not defined at all